File: Labtwo/JAVA/pr1.java
import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class RegexFinder {
    /**
     * Метод для поиска всех вхождений паттерна в строке
     * @param str - строка для поиска
     * @param pattern - регулярное выражение
     * @return список найденных совпадений
     */
    public static void findPattern(String str, String regex) {
        // Компилируем регулярное выражение
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(str);

        // Поиск всех совпадений
        boolean found = false;
        while (matcher.find()) {
            System.out.println("Found: " + matcher.group());
            found = true;
        }

        if (!found) {
            System.out.println("No matches found.");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Стандартный паттерн для поиска (1?10+1)
        String defaultPattern = "1?10+1";
        
        System.out.println("Enter your string (pattern to search: " + defaultPattern + "):");
        String input = scanner.nextLine();

        System.out.println("Matching results:");
        findPattern(input, defaultPattern);
        
        scanner.close();
    }
}

File: Labtwo/KOTLIN/pr1.kt
import java.util.regex.Pattern

// Функция поиска совпадений по регулярному выражению
fun findPattern(input: String, pattern: Pattern): List<String> {
    val matcher = pattern.matcher(input)
    val results = mutableListOf<String>()
    // Ищем все совпадения
    while (matcher.find()) {
        results.add(matcher.group())
    }
    return results
}

fun main() {
    // Компилируем регулярное выражение
    val pattern = Pattern.compile("1?10+1")
    
    println("Write your string->")
    val input = readLine() ?: ""
    
    // Получаем и выводим результаты
    val results = findPattern(input, pattern)
    results.forEach { println(it) }
}